applications/Christensen.cc

```diff
@@ -460,7 +454,10 @@ void Christensen<dim>::run()
   temperature->solution = temperature->old_solution;
   output();
 
-  while (time_stepping.get_current_time() < time_stepping.get_end_time())
+  const unsigned int n_steps = this->prm.time_discretization_parameters.n_maximum_steps;
+
+  while (time_stepping.get_current_time() < time_stepping.get_end_time() &&
+         time_stepping.get_step_number() < n_steps)
```

Comment:
Could a bypass case be introduced? For example, when `.n_maximum_steps == 0` it is ignored. This avoids having to compute the number of steps based upon the time step and end time values for each benchmark.

Reply:
I implemented this in 9cb0306ced3c70b7d825aa858a8dc03f0e8c9924